### DISCUSSION TITLE
Welcome to peripage-kotlin-bluetooth-printer Discussions!

### DISCUSSION THREAD

**Discussion by Dibyakshu** [Dibyakshu/peripage-kotlin-bluetooth-printer#1] [Announcements]
## 👋 Welcome!
  We’re using Discussions as a place to connect with other members of our community. We hope that you:
  * Ask questions you’re wondering about.
  * Share ideas.
  * Engage with other community members.
  * Welcome others and are open-minded. Remember that this is a community we
  build together 💪.

  To get started, comment below with an introduction of yourself and tell us about what you do with this community.

**Comment by malik-palmpower** [comment]
How to print a4 sheet pdf document using this library ?

**Reply by Dibyakshu** [reply]
Hi this library is for usage with peri page small thermal printers only. This code is not for a4 paper but for approx 5.5 cm thermal paper and around 384 pixels wide. 

However if u want to build something for a4 paper, u can change the scale instead of using 384 pixels 3508 pixels for 300 dpi. 

This library can be used for u to understand how to connect a bluetooth device and how to print by converting image to byte array to print. Also that we need to send commands for bluetooth devices to do certain functions. Example 0x0001AF062.

But all Bluetooth devices use different commands to print so the command u need to send to printer to print might vary.

This can be a good base to start writing code for your a4 printer.

**Comment by malik-palmpower** [comment]
Thank you for your reply.

On Wed, Jan 31, 2024 at 12:50 PM Dibyakshu Basumatary <
***@***.***> wrote:

> Hi this library is for usage with peri page small thermal printers only.
> This code is not for a4 paper but for approx 5.5 cm thermal paper and
> around 384 pixels wide.
>
> However if u want to build something for a4 paper, u can change the scale
> instead of using 384 pixels 3508 pixels for 300 dpi.
>
> This library can be used for u to understand how to connect a bluetooth
> device and how to print by converting image to byte array to print. Also
> that we need to send commands for bluetooth devices to do certain
> functions. Example 0x0001AF062.
>
> But all Bluetooth devices use different commands to print so the command u
> need to send to printer to print might vary.
>
> This can be a good base to start writing code for your a4 printer.
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/Dibyakshu/peripage-kotlin-bluetooth-printer/discussions/1#discussioncomment-8315756>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/BFRL426E2H3F527UDT4C6I3YRHWDTAVCNFSM6AAAAAA6RSLF7OVHI2DSMVQWIX3LMV43SRDJONRXK43TNFXW4Q3PNVWWK3TUHM4DGMJVG42TM>
> .
> You are receiving this because you commented.Message ID:
> <Dibyakshu/peripage-kotlin-bluetooth-printer/repo-discussions/1/comments/8315756
> @github.com>
>


-- 
Thanks & Regards,

Muhammed Malik
